packages/react-vapor/src/components/listBox/ListBox.tsx

```diff
@@ -54,39 +59,45 @@ export class ListBox extends React.Component<IListBoxProps, {}> {
         const visibleLength = _.filter(this.props.items, (item: IItemBoxProps) => shouldShow(item) && !item.disabled)
             .length;
 
-        let index = 0;
+        let realIndex = 0;
         let activeSet = false;
         const items = _.chain(this.props.items)
             .filter(shouldShow)
             .map((item: IItemBoxProps) => {
                 let active = false;
+                const itemWithIndex: IITemBoxPropsWithIndex = {item: {...item}};
                 if (!item.disabled) {
                     if (this.props.active === null) {
                         active = _.contains(this.props.selected, item.value);
                     } else {
-                        active = mod(this.props.active, visibleLength) === index;
+                        active = mod(this.props.active, visibleLength) === realIndex;
                     }
                     activeSet = active || activeSet;
-                    index++;
+
+                    itemWithIndex.index = realIndex;
+                    itemWithIndex.item.active = active;
+                    realIndex++;
                 }
-                return {...item, active};
+                return itemWithIndex;
             })
-            .map((item: IItemBoxProps) => {
+            .map(({item, index}: IITemBoxPropsWithIndex) => {
                 let active = item.active;
                 if (!item.disabled && activeSet === false) {
                     active = true;
                     activeSet = true;
                 }
-                return {...item, active};
+                return {item: {...item, active}, index};
+            })

```

Comment:
```typescript
.map((itemWithIndex: IITemBoxPropsWithIndex) => {
    let active = itemWithIndex.active;
    if (!itemWithIndex.disabled && activeSet === false) {
        active = true;
        activeSet = true;
    }
    return {...itemWithIndex, active};
})
```

Reply:
So *itemWithIndex* has the following props: **item**, **index** 

In your suggestion, active would be a new prop to *itemWithIndex* instead of *item*.

Reply:
I could probably use something like that:
`return _.merge(itemWithIndex, {item: {active}}}`